pkg/grpc_middleware/dialer/builder.go

```diff
@@ -0,0 +1,319 @@
+// connection. We leverage grpc_middleware package ChainUnaryClient to creates a
+// single interceptor out of a chain of many interceptors to override the grpc
+// default behavior of only allowing one.
+func (b *Builder) WithUnaryInterceptors(interceptors ...grpc.UnaryClientInterceptor) {
+	b.uinterceptors = interceptors
+}
+
+// AppendUnaryInterceptors ...
+func (b *Builder) AppendUnaryInterceptors(interceptors ...grpc.UnaryClientInterceptor) {
+	b.uinterceptors = append(b.uinterceptors, interceptors...)
+}
+
+// GetUnaryInterceptors returns the UnaryClientInterceptors slice
+func (b *Builder) GetUnaryInterceptors() []grpc.UnaryClientInterceptor {
+	return b.uinterceptors
+}
+
+// WithStreamInterceptors set a list of interceptors to the Grpc client for unary
+// connection. We leverage grpc_middleware package ChainStreamClient to creates a
+// single interceptor out of a chain of many interceptors to override the grpc
+// default behavior of only allowing one.
+func (b *Builder) WithStreamInterceptors(interceptors ...grpc.StreamClientInterceptor) {
+	b.sinterceptors = interceptors
+}
+
+// AppendStreamInterceptors ...
+func (b *Builder) AppendStreamInterceptors(interceptors ...grpc.StreamClientInterceptor) {
+	b.sinterceptors = append(b.sinterceptors, interceptors...)
+}
+
+// GetStreamInterceptors returns the StreamClientInterceptors slice
+func (b *Builder) GetStreamInterceptors() []grpc.StreamClientInterceptor {
+	return b.sinterceptors
+}
+
+// WithServerTransportCredentials builds transport credentials for the service a gRPC client connects with
+func (b *Builder) WithServerTransportCredentials(insecureSkipVerify bool, certPool *x509.CertPool) {
+	if b.tlsConfig == nil {
+		b.tlsConfig = &tls.Config{}
+	}
+	if insecureSkipVerify {
+		b.tlsConfig.InsecureSkipVerify = true
+	}
+	if certPool != nil {
+		b.tlsConfig.RootCAs = certPool
+	}
+}
+
+// GetServerTransportCredentials ...
+func (b *Builder) GetServerTransportCredentials() (bool, *x509.CertPool) {
+	return b.tlsConfig.InsecureSkipVerify, b.tlsConfig.RootCAs
+}
+
+// WithClientTransportCredentials builds transport credentials for a gRPC client uses to connect to service
+func (b *Builder) WithClientTransportCredentials(certs ...tls.Certificate) {
+	b.tlsConfig.Certificates = certs
+}
+
+// GetClientTransportCredentials ...
+func (b *Builder) GetClientTransportCredentials() []tls.Certificate {
+	return b.tlsConfig.Certificates
+}
+
+// WithClientCredentials builds transport credentials for a gRPC client
+func (b *Builder) WithClientCredentials(c credentials.PerRPCCredentials) {
+	b.credentials = c
+}
+
+// GetClientCredentials ...
+func (b *Builder) GetClientCredentials() credentials.PerRPCCredentials {
+	return b.credentials
+}
+
+// SetConnInfo allows passing in the dns and port for the connection, providing
+// flexibility if a consumer wants to set a default and or override
+func (b *Builder) SetConnInfo(dns, port string, tls bool) error {
```

Comment:
port shouldn't be a string, should be `uint`

Reply:
or uint16 :)